File: src/main/java/br/com/zup/academy/dominio/repository/CategoriaRepository.java
package br.com.zup.academy.dominio.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import br.com.zup.academy.dominio.modelo.Categoria;

@Repository
public interface CategoriaRepository extends JpaRepository<Categoria, Long>{

	Optional<Categoria> findByNome(String nome);
}

File: casa-do-codigo-grpc/src/main/kotlin/br/com/zup/dominio/repository/CategoriaRepository.kt
package br.com.zup.dominio.repository

import br.com.zup.dominio.modelo.Categoria
import io.micronaut.data.annotation.Repository
import io.micronaut.data.jpa.repository.JpaRepository

@Repository
interface CategoriaRepository: JpaRepository<Categoria, Long> {

}
